components/Navbar.js

```diff
@@ -1,109 +1,90 @@
-	return (
-		<li id="committee-mobile-nav-item">
-			<ul className="committee-mobile-nav">
-				<li><Link href={'/committees#studio'}><a>Studio</a></Link></li>
-				<li><Link href={'/committees#icpc'}><a>ICPC</a></Link></li>
-				<li><Link href={'/committees#design'}><a>Design</a></Link></li>
-				<li><Link href={'/committees#cyber'}><a>Cyber</a></Link></li>
-				<li><Link href={'/committees#teachla'}><a>Teach LA</a></Link></li>
-				<li><Link href={'/committees#w'}><a>W</a></Link></li>
-				<li><Link href={'/committees#ai'}><a>AI</a></Link></li>
-				<li><Link href={'/committees#hack'}><a>Hack</a></Link></li>
-			</ul>
-		</li>
-	);
-}
+export default function Navbar() {
+  // set states
+  const [menuActive, setMenuActive] = useState('');
+  const [linkActive, setLinkActive] = useState('');
 
+  const router = useRouter();
 
-export default class Navbar extends React.Component {
-	constructor(props) {
-		super(props);
-		this.clickMobileNav = this.clickMobileNav.bind(this);
-	}
+  // switches mobile menu state
+  const menuActivate = () => {
+    setMenuActive(menuActive ? '' : 'active');
+  };
 
-	clickMobileNav() {
-		// TODO: remove direct DOM manipulation, make this stateful
-		document.getElementById('menu-toggle').checked = false;
-	}
+  // allows accessible access to mobile menu
+  const menuAccessible = (event) => {
+    if (event.code === 'Enter' || event.code === 'Space') {
+      setMenuActive(menuActive ? '' : 'active');
+    }
+  };
 
-	// TODO: can we both refactor this to use useEffect, and/or remove
-	// this listener entirely
+  useEffect(() => {
+    const hamburger = document.querySelector('#hamburger');
+    const navItems = document.querySelector('.nav-items');
+    const listItems = navItems.querySelectorAll(':scope > li');
 
-	componentDidMount() {
-		// TODO: remove direct DOM manipulation, make this stateful
-		const items = document.querySelectorAll('#mobile-nav .nav-items li');
-		for (const item of items) {
-			item.addEventListener('click', this.clickMobileNav);
-		}
-	}
+    setLinkActive(router.asPath);
+    hamburger.addEventListener('click', menuActivate);
+    hamburger.addEventListener('keyup', menuAccessible);
+    // closes menu on link click
+    listItems.forEach((item) => {
+      item.addEventListener('click', menuActivate);
+      item.addEventListener('keyup', menuAccessible);
+    });
 
-	componentWillUnmount() {
-		// TODO: remove direct DOM manipulation, make this stateful
-		const items = document.querySelectorAll('#mobile-nav .nav-items li');
-		for (const item of items) {
-			item.removeEventListener('click', this.clickMobileNav);
-		}
-	}
+    return function cleanUp() {
+      hamburger.removeEventListener('click', menuActivate);
+      hamburger.removeEventListener('keyup', menuAccessible);
+      listItems.forEach((item) => {
+        item.removeEventListener('click', menuActivate);
+        item.removeEventListener('keyup', menuAccessible);
+      });
+    };
+  }, [menuActive]);
```

Comment:
Oh, one final note - not sure if `menuActive` is needed here as a dependency, since I imagine we only want to add/remove listeners on page load/exit, not when the actual menu gets opened or closed? But I might be misunderstanding!

Reply:
We're calling a page render everytime menu active updates in order to close the menu on click. I didn't originally have it there but noticed during testing that the menu didn't close.